pyanaconda/kickstart.py

```diff
@@ -669,16 +668,17 @@ def ksshowwarning(message, category, filename, lineno, file=None, line=None):
 
             # Set up the warnings module.
             warnings.showwarning = ksshowwarning
-
-            for category in kscategories:
-                warnings.filterwarnings(action="always", module=ksmodule, category=category)
+            warnings.filterwarnings(action="always", module=ksmodule,
+                                    category=KickstartParseWarning)
 
             # Parse the kickstart file in DBus modules.
             if pass_to_boss:
                 boss = BOSS.get_proxy()
                 report = KickstartReport.from_structure(
                     boss.ReadKickstartFile(f)
                 )
+                for warn in report.warning_messages:
+                    warnings.warn(warn.message)
```

Comment:
Please, set the category to `KickstartParseWarning`, otherwise these warnings will not be collected in `kswarnings`.